pkg/entrypoint/admissioncontroller/admissioncontroller.go

```diff
@@ -0,0 +1,217 @@
+	protectedSCCs     map[string]security.SecurityContextConstraints
+}
+
+func (ac *admissionController) getAdmissionReviewRequest(r *http.Request) (req *admissionv1beta1.AdmissionRequest, errorcode int) {
+	ac.log.Printf("New review request %s", r.RequestURI)
+	if r.Method != http.MethodPost {
+		return nil, http.StatusMethodNotAllowed
+	}
+	if r.Header.Get("Content-Type") != "application/json" {
+		return nil, http.StatusUnsupportedMediaType
+	}
+
+	var reviewIncoming *admissionv1beta1.AdmissionReview
+	err := json.NewDecoder(r.Body).Decode(&reviewIncoming)
+	if err != nil {
+		return nil, http.StatusBadRequest
+	}
+	req = reviewIncoming.Request
+	return req, 0
+}
+
+func (ac *admissionController) run() error {
+	ac.protectedSCCs = ac.InitProtectedSCCs()
+	mux := &http.ServeMux{}
+	mux.HandleFunc("/pods", ac.handleWhitelist)
+	mux.HandleFunc("/daemonsets", ac.handleWhitelist)
+	mux.HandleFunc("/replicasets", ac.handleWhitelist)
+	mux.HandleFunc("/statefulsets", ac.handleWhitelist)
+	mux.HandleFunc("/jobs", ac.handleWhitelist)
+	mux.HandleFunc("/cronjobs", ac.handleWhitelist)
+	mux.HandleFunc("/deploymentconfigs", ac.handleWhitelist)
+	mux.HandleFunc("/deployments", ac.handleWhitelist)
+	mux.HandleFunc("/sccs", ac.handleSCC)
+
+	mux.HandleFunc("/healthz", ac.handleHealthz)
+	mux.HandleFunc("/healthz/ready", ac.handleHealthz)
+
+	ac.log.Print("Aro Admission Controller starting.")
+	err := http.ListenAndServeTLS(":8443", "/etc/aro-admission-controller/aro-admission-controller.crt", "/etc/aro-admission-controller/aro-admission-controller.key", mux)
+	if err != nil {
+		ac.log.Fatal("ListenAndServeTLS: ", err)
+	}
+	return err
+}
+
+func getRestrictedSCC() (*security.SecurityContextConstraints, error) {
+	var restricted *security.SecurityContextConstraints
+
+	groups, users := bootstrappolicy.GetBoostrapSCCAccess(bootstrappolicy.DefaultOpenShiftInfraNamespace)
+	for _, scc := range bootstrappolicy.GetBootstrapSecurityContextConstraints(groups, users) {
+		if scc.Name == bootstrappolicy.SecurityContextConstraintRestricted {
+			restricted = scc
+		}
+	}
+	if restricted == nil {
+		return nil, fmt.Errorf("couldn't find restricted SCC in bootstrappolicy")
+	}
+
+	return restricted, nil
+}
+
+func start(cfg *cmdConfig) error {
+	var c admissionControllerConfig
+
+	logrus.SetLevel(log.SanitizeLogLevel(cfg.LogLevel))
+	logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+	log := logrus.NewEntry(logrus.StandardLogger())
+
+	var whitelistedImages = []*regexp.Regexp{}
+	c.loadConfig(log)
+	for _, w := range c.Whitelist {
+		whitelistedImages = append(whitelistedImages, regexp.MustCompile(w))
+	}
+	restricted, err := getRestrictedSCC()
+	if err != nil {
+		return err
+	}
+
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+
+	restconfig, err := kubeconfig.ClientConfig()
+	if err != nil {
+		restconfig, err = rest.InClusterConfig()
```

Comment:
do you actually ever use InClusterConfig?  If not, remove

Reply:
Removed.